fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimerTest.java

```diff
@@ -0,0 +1,192 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.KeyValue;
+import com.apple.foundationdb.async.MoreAsyncUtil;
+import com.apple.foundationdb.record.ExecuteProperties;
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.RecordCursorResult;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TupleRange;
+import com.apple.foundationdb.record.provider.common.StoreTimer;
+import com.apple.foundationdb.record.provider.common.StoreTimerSnapshot;
+import com.apple.foundationdb.record.provider.foundationdb.keyspace.KeySpacePath;
+import com.apple.foundationdb.subspace.Subspace;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.Tags;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tests for {@link FDBStoreTimer}.
+ */
+@Tag(Tags.RequiresFDB)
+public class FDBStoreTimerTest {
+    FDBDatabase fdb;
+    FDBRecordContext context;
+    private Subspace subspace;
+    ExecuteProperties ep;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        fdb = FDBDatabaseFactory.instance().getDatabase();
+        context = fdb.openContext();
+        setupBaseData();
+        FDBStoreTimer timer = new FDBStoreTimer();
+        context.setTimer(timer);
+    }
+
+    @AfterEach
+    public void teardown() throws Exception {
+        context.close();
+        fdb.close();
+    }
+
+    @Test
+    public void counterDifferenceTest() {
+        RecordCursor<KeyValue> kvc = KeyValueCursor.Builder.withSubspace(subspace).setContext(context).setScanProperties(ScanProperties.FORWARD_SCAN).setRange(TupleRange.ALL).build();
+
+        //see the timer counts from some onNext calls
+        FDBStoreTimer latestTimer = context.getTimer();
+        StoreTimerSnapshot savedTimer;
+        CompletableFuture<RecordCursorResult<KeyValue>> fkvr;
+        RecordCursorResult<KeyValue> kvr;
+        StoreTimer diffTimer;
+        Map<String, Number> diffKVs;
+
+        //get a snapshot from latestTimer before advancing cursor
+        savedTimer = StoreTimerSnapshot.from(latestTimer);
+
+        //advance the cursor once
+        kvr = kvc.onNext().join();
+
+        // the diff from latestTimer minus savedTimer will have the timer cost from the single cursor advance
+        diffTimer = StoreTimer.subtractSnapshot(latestTimer, savedTimer);
+        diffKVs = diffTimer.getKeysAndValues();
+        Assertions.assertTrue(diffKVs.containsKey("load_scan_entry_count"));
+        Assertions.assertEquals(diffKVs.get("load_scan_entry_count").intValue(), 1);
```

Comment:
Is this a convention or correctness issue? 

Java Class: org.junit.Assert

Assert class provides a set of assertion methods useful for writing tests.

Assert.assertEquals() methods checks that the two objects are equals or not. If they are not, an AssertionError without a message is thrown. Incase if both expected and actual values are null, then this method returns equal. In the below example, the first Test (mySimpleEqualsTest()) compares two strings. The second test (myObjectEqualsTest()) we are comparing two different user defined objects. The assertEquals() method calls equals method on each object to check equality.

Reply:
It's, like, in between. If you do it the other way, you're asserts will still fire at exactly the right time, but if there are problems, the error message is backwards. It will say "expected<wrong value> but was <right value>".